### PR TITLE
refactor: extract QrCodeService

### PR DESCRIPTION
**Pull request by kylerwsm**
## Problem

This PR contribute towards refactoring our backend to adopt the controller service repository pattern. Specifically, in this PR, I extracted QrCodeService from our /qrcode endpoint.

### CODE REVIEW

**Comment by kylerwsm** [comment]
Addressed merge conflicts

File: src/server/services/QrCodeService/index.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      margin: 0,
+      errorCorrectionLevel: 'H',
+      color: {
+        dark: '#384A51',
+      },
+    })
+  }
+
+  // Build QR code string with GoGovSg logo.
+  public makeGoQrCode: (url: string) => Promise<[Buffer, number]> = async (
```
Comment (liangyuanruo) [inline src/server/services/QrCodeService/index.ts:39]:
```suggestion
  private makeGoQrCode: (url: string) => Promise<[Buffer, number]> = async (
```

File: src/server/services/QrCodeService/index.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+export const FONT_SIZE = 32
+export const LINE_HEIGHT = 1.35
+
+// Convert readFile callback function to promise function.
+const readFile = util.promisify(fs.readFile)
+
+@injectable()
+export class QrCodeService implements QrCodeServiceInterface {
+  // Build base QR code string without logo.
+  public makeQrCode: (url: string) => Promise<string> = (url) => {
```
Comment (liangyuanruo) [inline src/server/services/QrCodeService/index.ts:27]:
```suggestion
  private makeQrCode: (url: string) => Promise<string> = (url) => {
```

**Comment by kylerwsm** [comment]
Suggestions accepted and passed CI

**Comment by LoneRifle** [comment]
Is this related to #156 ?

**Pull request merged**